guarddog/scanners/scanner.py

```diff
@@ -229,16 +229,17 @@ def scan_local(self, path, rules=None, callback: typing.Callable[[dict], None] =
         if rules is not None:
             rules = set(rules)
 
-        if os.path.exists(path):
-            if path.endswith('.tar.gz') or path.endswith('.tgz') or path.endswith('.zip') or path.endswith('.whl'):
-                with tempfile.TemporaryDirectory() as tmpdirname:
-                    safe_extract(path, tmpdirname)
-                    return self.analyzer.analyze_sourcecode(tmpdirname, rules=rules)
-            elif os.path.isdir(path):
-                return self.analyzer.analyze_sourcecode(path, rules=rules)
-            else:
-                raise Exception(f"Path {path} is not a directory nor an archive type supported by GuardDog.")
-        raise Exception(f"Path {path} does not exist.")
+        if not os.path.exists(path):
+            raise Exception(f"Path {path} does not exist.")
+
+        if any(path.endswith(ext) for ext in ('.tar.gz', '.tgz', '.zip', '.whl')):
+            with tempfile.TemporaryDirectory() as tmpdirname:
+                safe_extract(path, tmpdirname)
+                return self.analyzer.analyze_sourcecode(tmpdirname, rules=rules)
+        elif os.path.isdir(path):
```

Comment:
this elif too can be avoided since the last line is a return

```
if any(path.endswith(ext) for ext in ('.tar.gz', '.tgz', '.zip', '.whl')):
    ...
    return self.analyzer.analyze_sourcecode(tmpdirname, rules=rules)

if os.path.isdir(path):
   return ...

raise Exception(f"Path {path} is not a directory nor an archive type supported by GuardDog.")
```

Reply:
Yep, didn't catch that